rueidis.go

```diff
@@ -296,6 +301,12 @@ func NewClient(option ClientOption) (client Client, err error) {
 	}
 	pmbk := option.PipelineMultiplex
 	option.PipelineMultiplex = 0 // PipelineMultiplex is meaningless for cluster client
+
+	if option.ForceSingleClient {
+		option.PipelineMultiplex = singleClientMultiplex(pmbk)
+		conn := makeConn(option.InitAddress[0], &option)
+		return newSingleClientWithConn(conn, cmds.NewBuilder(cmds.NoSlot), !option.DisableRetry), nil
```

Comment:
```suggestion
		return newSingleClient(&option, nil, makeConn)
```
The `newSingleClientWithConn` will always return a nil error which is not good.